### PR TITLE
Train code refactoring, Add wandb

### PR DESCRIPTION
**Pull request by darthegg**
Train code refactoring, Add wandb

### CODE REVIEW

File: run_CartPole_v0.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        score += reward
+
+        # Epsilon decay
+        epsilon = agent.decay_epsilon(epsilon, H["MIN_EPSILON"], H["MAX_EPSILON"], H["EPSILON_DECAY"])
+
+        # Update target network
+        if total_step >= H["TARGET_UPDATE_AFTER"] and total_step % H["TARGET_UPDATE_STEP"] == 0:
+            model_target.load_state_dict(model.state_dict())
+
+        if done is True:
```
Comment (whikwon) [inline run_CartPole_v0.py]: https://www.python.org/dev/peps/pep-0008/

Don't compare boolean values to True or False using ==.
```
Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```

File: agent/dqn.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        self.env = env
+        self.model = model
+        self.model_target = model_target
+        self.optimizer = optimizer
+        self.memory = memory
+        self.gamma = gamma
+        self.device = device
+
+    def select_action(self, state: np.ndarray, epsilon: float, is_learn: bool = True) -> int:
```
Comment (whikwon) [inline agent/dqn.py]: 주로 train, test 단어를 많이 사용하는데 learn을 사용하신 이유가 있을까요

File: model/dqn.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- Reference: https://storage.googleapis.com/deepmind-media/dqn/DQNNaturePaper.pdf (DQN)
+             https://github.com/medipixel/rl_algorithms (Medipixel Github)
+             https://github.com/higgsfield/RL-Adventure (RL Adventure)
+"""
+
+import torch.nn as nn
+import gym
+
+
+class DQN(nn.Module):
```
Comment (whikwon) [inline model/dqn.py]: super class인 `nn.Module` `def forward`가 구현이 되어 있어서 상속 받아서 사용할 때 `__init__` 시에 사용 할 module 들을 만들고 얘를 `def forward`에서 가져다가 활용하는 식으로 많이 구현합니다. `def forward`가 구현되지 않으면 에러가 발생하게 끔 되어 있는데 에러 발생 안 하셨었나요?

https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html

File: agent/dqn.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        self.optimizer.step()
+        return loss.item()
+
+    def write_log(self, episode: int, score: float, loss: float, epsilon: float, is_log: bool):
+        print(
+            f"episode : {episode}\tscore : {score}",
+            "\tLoss : %.4f" % (loss),
+            "\tEpsilon : %.4f" % (epsilon))
+
+        if is_log is True:
```
Comment (whikwon) [inline agent/dqn.py]: https://www.python.org/dev/peps/pep-0008/

Don't compare boolean values to True or False using ==.
```
Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```

File: run_CartPole_v0.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+
+# Env initialize
+env = gym.make("CartPole-v0")
+
+# Seeding
+env.seed(2)
+torch.manual_seed(2)
+np.random.seed(2)
+random.seed(2)
```
Comment (whikwon) [inline run_CartPole_v0.py]: `random.seed`는 어디에 사용되나요?

File: agent/dqn.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        loss = self.get_td_loss()
+        self.optimizer.zero_grad()
+        loss.backward()
+        self.optimizer.step()
+        return loss.item()
+
+    def write_log(self, episode: int, score: float, loss: float, epsilon: float, is_log: bool):
+        print(
+            f"episode : {episode}\tscore : {score}",
+            "\tLoss : %.4f" % (loss),
```
Comment (whikwon) [inline agent/dqn.py]: f-string과 다른 방식이 혼재되어 있습니다. 
https://docs.python.org/ko/3/tutorial/inputoutput.html

**Comment by darthegg** [comment]
1. Linting
2. Test code
3. Baseline
